File: medium/1318. Minimum Flips to Make a OR b Equal to c/1318. Minimum Flips to Make a OR b Equal to c.go
package main

func minFlips(a int, b int, c int) int {
	flips := 0

	for a != 0 || b != 0 || c != 0 {
		ai, bi, ci := a%2, b%2, c%2
		a, b, c = a/2, b/2, c/2

		if ai|bi != ci {
			flips += 1
			if ci == 0 && ai == 1 && bi == 1 {
				flips += 1
			}
		}
	}

	return flips
}

File: medium/1318. Minimum Flips to Make a OR b Equal to c/1318. Minimum Flips to Make a OR b Equal to c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        flips = 0

        while a or b or c:
            if c % 2 == 1:
                flips += int(not (a % 2 or b % 2))
            else:
                flips += int(a % 2) + int(b % 2)

            a //= 2
            b //= 2
            c //= 2

        return flips


solution = Solution()
print(solution.minFlips(1, 4, 5))
